File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/ComplexMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.model.complex.BloodPressureObject
import com.rookmotion.rook.sdk.domain.model.complex.PositionObject
import com.rookmotion.rook.sdk.domain.model.complex.TemperatureObject
import com.rookmotion.rook.sdk.domain.model.complex.VelocityObject
import com.rookmotion.rook_sdk_health_connect.proto.BloodPressureObjectProto
import com.rookmotion.rook_sdk_health_connect.proto.PositionObjectProto
import com.rookmotion.rook_sdk_health_connect.proto.TemperatureObjectProto
import com.rookmotion.rook_sdk_health_connect.proto.VelocityObjectProto

fun BloodPressureObject.toObjectProto(): BloodPressureObjectProto {
    return BloodPressureObjectProto.newBuilder()
        .setSystolicMmHg(systolicMmHg)
        .setDiastolicMmHg(diastolicMmHg)
        .build()
}

fun PositionObject.toObjectProto(): PositionObjectProto {
    return PositionObjectProto.newBuilder()
        .setLatDeg(latDeg)
        .setLngDeg(lngDeg)
        .build()
}

fun TemperatureObject.toObjectProto(): TemperatureObjectProto {
    return TemperatureObjectProto.newBuilder()
        .setTemperatureCelsius(temperatureCelsius)
        .setMeasurementType(measurementType)
        .build()
}

fun VelocityObject.toObjectProto(): VelocityObjectProto {
    return VelocityObjectProto.newBuilder()
        .setSpeedMetersPerSecond(speedMetersPerSecond)
        .setDirection(direction)
        .build()
}

File: packages/rook_sdk_apple_health/ios/Classes/Mapper/ComplexMappers.swift
//
//  ComplexMappers.swift
//  rook_sdk_apple_health
//
//  Created by Daniel Nolasco on 28/10/25.
//

import Foundation
import RookSDK

extension RookTemperature {
    func toObjectProto() -> TemperatureObjectProto {
        return TemperatureObjectProto.with {
            $0.measurementType = measurementType
            $0.temperatureCelsius = temperatureCelsius.toDouble()
        }
    }
}

extension RookBloodPressureSystolicDiastolic {
    func toObjectProto() -> BloodPressureObjectProto {
        return BloodPressureObjectProto.with {
            $0.systolicMmHg = systolicBp.toDouble()
            $0.diastolicMmHg = diastolicBp.toDouble()
        }
    }
}

extension RookVelocityVectorSpeed {
    func toObjectProto() -> VelocityObjectProto {
        return VelocityObjectProto.with {
            $0.speedMetersPerSecond = speedMetersPerSecond.toDouble()
            $0.direction = direction
        }
    }
}

extension RookPositionLatLng {
    func toObjectProto() -> PositionObjectProto {
        return PositionObjectProto.with {
            $0.latDeg = lat
            $0.lngDeg = lng
        }
    }
}
